XamarinCommunityToolkit/Effects/Touch/PlatformTouchEffect.android.cs

```diff
@@ -0,0 +1,340 @@
+			}
+		}
+
+		protected override void OnDetached()
+		{
+			if (effect?.Control == null)
+				return;
+
+			try
+			{
+				if (accessibilityManager != null)
+				{
+					accessibilityManager.RemoveAccessibilityStateChangeListener(accessibilityListener);
+					accessibilityManager.RemoveTouchExplorationStateChangeListener(accessibilityListener);
+					accessibilityListener.Dispose();
+					accessibilityManager = null;
+					accessibilityManager = null;
+				}
+
+				if (View != null)
+				{
+					View.LayoutChange -= LayoutChange;
+					View.Touch -= OnTouch;
+					View.Click -= OnClick;
+				}
+
+				effect.Control = null;
+				effect = null;
+
+				if (viewOverlay != null)
+				{
+					if (Group != null)
+						Group.RemoveView(viewOverlay);
+
+					viewOverlay.Pressed = false;
+					viewOverlay.Foreground = null;
+					viewOverlay.Dispose();
+					ripple?.Dispose();
+				}
+			}
+			catch (ObjectDisposedException)
+			{
+				// Suppress exception
+			}
+			isHoverSupported = false;
+		}
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+			if (args.PropertyName == TouchEffect.IsAvailableProperty.PropertyName ||
+				args.PropertyName == VisualElement.IsEnabledProperty.PropertyName)
+			{
+				UpdateClickHandler();
+			}
+		}
+
+		void UpdateClickHandler()
+		{
+			View.Click -= OnClick;
+			if (IsAccessibilityMode || (effect.IsAvailable && effect.Control.IsEnabled))
+			{
+				View.Click += OnClick;
+				return;
+			}
+		}
+
+		void OnTouch(object sender, AView.TouchEventArgs e)
+		{
+			e.Handled = false;
+
+			if (effect?.IsDisabled ?? true)
+                return;
+
+			if (IsAccessibilityMode)
+				return;
+
+			switch (e.Event.ActionMasked)
+			{
+				case MotionEventActions.Down:
+					IsCanceled = false;
+					startX = e.Event.GetX();
+					startY = e.Event.GetY();
+					effect?.HandleUserInteraction(TouchInteractionStatus.Started);
+					effect?.HandleTouch(TouchStatus.Started);
+					StartRipple(e.Event.GetX(), e.Event.GetY());
+					if (effect.DisallowTouchThreshold > 0)
+						Group.Parent?.RequestDisallowInterceptTouchEvent(true);
+
+					break;
+				case MotionEventActions.Up:
+					HandleEnd(effect.Status == TouchStatus.Started ? TouchStatus.Completed : TouchStatus.Canceled);
+					break;
+				case MotionEventActions.Cancel:
+					HandleEnd(TouchStatus.Canceled);
+					break;
+				case MotionEventActions.Move:
+					if (IsCanceled)
+						return;
+
+					var diffX = Math.Abs(e.Event.GetX() - startX) / View.Context.Resources.DisplayMetrics.Density;
+					var diffY = Math.Abs(e.Event.GetY() - startY) / View.Context.Resources.DisplayMetrics.Density;
+					var maxDiff = Math.Max(diffX, diffY);
+					var disallowTouchThreshold = effect.DisallowTouchThreshold;
+					if (disallowTouchThreshold > 0 && maxDiff > disallowTouchThreshold)
+					{
+						HandleEnd(TouchStatus.Canceled);
+						return;
+					}
+					var view = sender as AView;
```

Comment:
Just to double check.
Are we sure that this value will never be `null`?

Reply:
Yeah, 100% cuz we subscribe it as ```View.Touch += OnTouch;``` where Views is AView